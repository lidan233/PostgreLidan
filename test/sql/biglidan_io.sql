-- simple input
SELECT '120'::biglidan;
SELECT '3c'::biglidan;
-- case insensitivity
SELECT '3C'::biglidan;
SELECT 'FoO'::biglidan;
-- invalid characters
SELECT 'foo bar'::biglidan;
SELECT 'abc$%2'::biglidan;
-- negative values
SELECT '-10'::biglidan;
-- to big values
SELECT 'abcdefghijklmn'::biglidan;

-- storage
BEGIN;
CREATE TABLE lidan_test(val biglidan);
INSERT INTO lidan_test VALUES ('123'), ('3c'), ('5A'), ('zZz');
SELECT * FROM lidan_test;
UPDATE lidan_test SET val = '567a' where val = '123';
SELECT * FROM lidan_test;
UPDATE lidan_test SET val = '-aa' where val = '3c';
SELECT * FROM lidan_test;
ROLLBACK;
