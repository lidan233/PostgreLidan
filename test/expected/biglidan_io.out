-- simple input
SELECT '120'::biglidan;
 biglidan 
-----------
 120
(1 row)

SELECT '3c'::biglidan;
 biglidan 
-----------
 3c
(1 row)

-- case insensitivity
SELECT '3C'::biglidan;
 biglidan 
-----------
 3c
(1 row)

SELECT 'FoO'::biglidan;
 biglidan 
-----------
 foo
(1 row)

-- invalid characters
SELECT 'foo bar'::biglidan;
ERROR:  invalid input syntax for biglidan: "foo bar"
LINE 1: SELECT 'foo bar'::biglidan;
               ^
SELECT 'abc$%2'::biglidan;
ERROR:  invalid input syntax for biglidan: "abc$%2"
LINE 1: SELECT 'abc$%2'::biglidan;
               ^
-- negative values
SELECT '-10'::biglidan;
 biglidan 
-----------
 -10
(1 row)

-- to big values
SELECT 'abcdefghijklmn'::biglidan;
ERROR:  value "abcdefghijklmn" is out of range for type biglidan
LINE 1: SELECT 'abcdefghijklmn'::biglidan;
               ^
-- storage
BEGIN;
CREATE TABLE lidan_test(val biglidan);
INSERT INTO lidan_test VALUES ('123'), ('3c'), ('5A'), ('zZz');
SELECT * FROM lidan_test;
 val 
-----
 123
 3c
 5a
 zzz
(4 rows)

UPDATE lidan_test SET val = '567a' where val = '123';
SELECT * FROM lidan_test;
 val  
------
 3c
 5a
 zzz
 567a
(4 rows)

UPDATE lidan_test SET val = '-aa' where val = '3c';
SELECT * FROM lidan_test;
 val  
------
 5a
 zzz
 567a
 -aa
(4 rows)

ROLLBACK;
